### PR TITLE
Add functions for service utility

### PR DESCRIPTION
**Pull request by aethyrxeo**

### CODE REVIEW

File: Test/Utils/ComputeNode/Service.ps1 [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    $NSSMServiceAlreadyCreatedError = 5
+    if ($Output.ExitCode -eq 0) {
+        Write-Log $Output.Output
+    }
+    elseif ($Output.ExitCode -eq $NSSMServiceAlreadyCreatedError) {
+        Write-Log "$ServiceName service already created, continuing..."
+    }
+    else {
+        throw [HardError]::new("Unknown (wild) error appeared while creating $ServiceName service")
```
Comment (sodar) [inline Test/Utils/ComputeNode/Service.ps1]: Let's put `$Output` and exit code in the exception message.

File: Test/Utils/ComputeNode/Service.ps1 [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }
+}
+
+function Out-StdoutAndStderrToLogFile {
+    Param (
+        [Parameter(Mandatory=$true)] $Session,
+        [Parameter(Mandatory=$true)] $ServiceName,
+        [Parameter(Mandatory=$true)] $LogPath
+    )
+    #redirect stdout and stderr to the log file
```
Comment (sodar) [inline Test/Utils/ComputeNode/Service.ps1]: This comment is not needed here. Function in self-explanatory.

File: Test/Utils/ComputeNode/Service.ps1 [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    )
+
+    $Output = Invoke-NativeCommand -Session $Session {
+        nssm remove $using:ServiceName confirm
+    } -CaptureOutput
+
+    Write-Log $Output
+}
+
+function Enable-Service {
```
Comment (sodar) [inline Test/Utils/ComputeNode/Service.ps1]: Enable/Disable is not a good name in my opinion. E.g. on systemd Enable/disable is used to signify if service is started automatically.

Maybe `Enable-Service` should be named `Start-RemoteService`. The same applies to `Disable-Service`.

**Comment by aethyrxeo** [comment]
This functions are used in [Create CNM Plugin service with NSSM](https://github.com/Juniper/contrail-windows-ci/pull/330) and [Create Agent Service with NSSM](https://github.com/Juniper/contrail-windows-ci/pull/340)

File: Test/Utils/ComputeNode/Service.ps1 [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+function Stop-RemoteService {
+    Param (
+        [Parameter(Mandatory=$true)] $Session,
+        [Parameter(Mandatory=$true)] $ServiceName
+    )
+
+    Invoke-Command -Session $Session -ScriptBlock {
+        Stop-Service $using:ServiceName -ErrorAction SilentlyContinue
```
Comment (m-kostrzewa) [inline Test/Utils/ComputeNode/Service.ps1]: Why is it ok to `ErrorAction SilentlyContinue` but not in `Start-RemoteService`?

File: Test/Utils/ComputeNode/Service.ps1 [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+function Out-StdoutAndStderrToLogFile {
+    Param (
+        [Parameter(Mandatory=$true)] $Session,
+        [Parameter(Mandatory=$true)] $ServiceName,
+        [Parameter(Mandatory=$true)] $LogPath
+    )
+
+    $Output = Invoke-NativeCommand -Session $Session -ScriptBlock {
+        nssm set $using:ServiceName AppStdout $using:LogPath
+        nssm set $using:ServiceName AppStderr $using:LogPath
```
Comment (jablonskim) [inline Test/Utils/ComputeNode/Service.ps1]: Does NSSM handle this situation (stdout and stderr redirected to single file) correctly?

Comment (aethyrxeo) [reply Test/Utils/ComputeNode/Service.ps1]: From what i saw, yes.

However, there is a case when deleting service and creating new one, that the old log file will be overwritten.

**Pull request merged**